packages/@aws-cdk/aws-bedrock-alpha/bedrock/agents/agent.ts

```diff
@@ -0,0 +1,769 @@
+  }
+
+  /**
+   * Render the agent collaborators.
+   *
+   * @internal This is an internal core function and should not be called directly.
+   */
+  private renderAgentCollaborators(): bedrock.CfnAgent.AgentCollaboratorProperty[] | undefined {
+    if (!this.agentCollaborators || this.agentCollaborators.length === 0) {
+      return undefined;
+    }
+
+    const agentCollaboratorsCfn: bedrock.CfnAgent.AgentCollaboratorProperty[] = [];
+
+    this.agentCollaborators.forEach(ac => {
+      agentCollaboratorsCfn.push(ac._render());
+    });
+    return agentCollaboratorsCfn;
+  }
+
+  /**
+   * Render the custom orchestration.
+   *
+   * @internal This is an internal core function and should not be called directly.
+   */
+  private renderCustomOrchestration(): bedrock.CfnAgent.CustomOrchestrationProperty | undefined {
+    if (!this.customOrchestration) {
+      return undefined;
+    }
+
+    return {
+      executor: {
+        lambda: this.customOrchestration.executor.lambdaFunction.functionArn,
+      },
+    };
+  }
+
+  // ------------------------------------------------------
+  // Validators
+  // ------------------------------------------------------
+  /**
+   * Check if the action group is valid
+   */
+  private validateActionGroup = (actionGroup: AgentActionGroup) => {
+    let errors: string[] = [];
+    // Find if there is a conflicting action group name
+    if (this.actionGroups?.find(ag => ag.name === actionGroup.name)) {
+      errors.push('Action group already exists');
+    }
+    return errors;
+  };
+
+  private generatePhysicalNameHash(
+    scope: IConstruct,
+    prefix: string,
+    options?: {
+      maxLength?: number;
+      lower?: boolean;
+      separator?: string;
+      allowedSpecialCharacters?: string;
+      destroyCreate?: any;
+    },
+  ): string {
+    const objectToHash = (obj: any): string => {
+      if (obj === undefined) { return ''; }
+      const jsonString = JSON.stringify(obj);
+      const hash = crypto.createHash('sha256');
+      return hash.update(jsonString).digest('hex').slice(0, 7);
+    };
+
+    const {
+      maxLength = 256,
+      lower = false,
+      separator = '',
+      allowedSpecialCharacters = undefined,
+      destroyCreate = undefined,
+    } = options ?? {};
+
+    const hash = objectToHash(destroyCreate);
+    if (maxLength < (prefix + hash + separator).length) {
+      throw new ValidationError('The prefix is longer than the maximum length.', this);
+    }
+
+    const uniqueName = Names.uniqueResourceName(
+      scope,
+      { maxLength: maxLength - (prefix + hash + separator).length, separator, allowedSpecialCharacters },
+    );
+    const name = `${prefix}${hash}${separator}${uniqueName}`;
+    if (name.length > maxLength) {
+      throw new ValidationError(`The generated name is longer than the maximum length of ${maxLength}`, this);
+    }
+    return lower ? name.toLowerCase() : name;
+  }
+
+  protected generatePhysicalName(): string {
```

Comment:
why do you overwrite the `generatePhysicalName` function that defined in Resource class. https://github.com/aws/aws-cdk/blob/fdc849245acffb34924c43d33f5c6d625ed34c19/packages/aws-cdk-lib/core/lib/resource.ts#L238-L240